.github/workflows/test.yml

```
@@ -0,0 +1,35 @@
+name: End-to-end tests
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Update
+        run: sudo apt update
+      - name: Install OS dependencies
+        run: sudo apt install -y python3-dev
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Start-up local test environment
+        uses: isbang/compose-action@v1.2.0
```

From https://github.com/isbang/compose-action#compose-file, we probably need to specify the right compose file to use?

```suggestion
        uses: isbang/compose-action@v1.2.0
        with:
          compose-file: "./ci-compose.yml"
```

---

(this might also fix the error from the failed run)

---

Thanks @daeMOn63, good catch!